src/lib/components/Error/ErrorReporter.tsx

```diff
@@ -0,0 +1,23 @@
+import { useAtomValue } from 'jotai/utils'
+import { providerAtom } from 'lib/state/web3'
+import { useEffect } from 'react'
+import { EMPTY } from 'widgets-web3-react/empty'
+
+class IntegrationError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'INTEGRATION ERROR'
+  }
+}
+
+const missingProviderError = new IntegrationError('Missing provider')
+
+export default function ErrorReporter() {
+  const [connector] = useAtomValue(providerAtom)
+  useEffect(() => {
+    if (connector === EMPTY) {
```

Comment:
Doesn't this start as empty? Will this always throw?

Reply:
Looks like I was out of the loop on how this actually works. I was thinking EMPTY is used primarily as a filler value and the component would always render w/ the integrator's provider value first.